### PR TITLE
Cherrypick the official Blake3 support

### PR DESCRIPTION
**Pull request by congt**
It reverts our own implementation of Blake3 support (https://github.com/bazel-ios/bazel-buildfarm/commit/26eb32e639c478c1543f7be354d701e54ef54fee) and cherrypicks the official Blake3 support(https://github.com/bazelbuild/bazel-buildfarm/pull/1697)

Because the bazel-ios-fork is far behind the latest main, we have to make a temporary patch on external dependencies.

### CODE REVIEW

File: src/main/java/build/buildfarm/common/blake3/JniLoader.java [hunk 1]
```diff
@@ -12,63 +12,50 @@
...
-            } catch (IOException e2) {
-              throw e;
-            }
-          }
-          break;
-
-        default:
-          throw new AssertionError("switch statement out of sync with OS values");
+      if (isDarwin()) {
+        loadLibrary("main/native/libblake3_jni.dylib");
```
Comment (gyfelton) [inline src/main/java/build/buildfarm/common/blake3/JniLoader.java:47]: gonna assume even the name is diff, what's behind the dylib is probably the same as `main/native/libunix_jni.dylib`?

**Review by gyfelton — APPROVED** [review]
do we want to test what happens in staging if we deliberately break it? for example point to a dylib that does not exist? that way we are more confident that staing not breaking really means staging not breaking + we know what kind of metrics goes wrong when this is happening / what alarm is set off?
for ec2 migration, we did this on bootstrapping for example by deliberately break it to ensure our alarm works.

File: src/main/java/build/buildfarm/common/blake3/JniLoader.java (hunk 1 above)
Comment (congt) [reply src/main/java/build/buildfarm/common/blake3/JniLoader.java:47]: Yes, it's the same

File: src/main/java/build/buildfarm/worker/shard/Worker.java [hunk 2]
```diff
@@ -606,7 +608,8 @@ public void start() throws ConfigurationException, InterruptedException, IOExcep
     // Create the appropriate writer for the context
     CasWriter writer;
     if (!configs.getWorker().getCapabilities().isCas()) {
-      writer = new RemoteCasWriter(backplane.getWorkers(), workerStubs);
+      Retrier retrier = new Retrier(Backoff.sequential(5), Retrier.DEFAULT_IS_RETRIABLE);
```
Comment (justinseanmartin) [inline src/main/java/build/buildfarm/worker/shard/Worker.java:611]: This is interesting that there is some backoff being introduced here. Seems a bit orthogonal to Blake3, so seemed worth calling out.

**Review by justinseanmartin — APPROVED** [review]
Are there any implications on the CAS or AC storage worker disk layout? Will these still cache hit for everything already populated, or will we have a bunch of redundant data in the cache until the old stuff expires?
